File: java/JavaFileutil/src/JavaFileutil/JavaFileCommands/DeleteCommand.java
package JavaFileutil.JavaFileCommands;

import JavaFileutil.AbstractCommands.AbstractUnaryFileCommand;
import JavaFileutil.Validators.FileExistenceValidator;

import java.util.List;

public class DeleteCommand extends AbstractUnaryFileCommand
{
    private static final String FileExistenceMessage = "cannot delete the file that doesn't exist.";
    public static final String UsageString =
        "delete command deletes <file1>.\n" +
        "Use it the following way:\n" +
        "\t'delete <file1>'\n" +
        "\t" + NotePreface + ArgumentCountMismatchMessage;

    private final void SetupValidation()
    {
        Validation.AddValidator(new FileExistenceValidator(Arguments, (short)0, FileExistenceMessage));
    }

    public DeleteCommand(List<String> args)
    {
        super(args);
        SetupValidation();
    }

    @Override
    public void Execute()
    {
        if (!Validation.Validate())
            return;

        try
        {
            file.delete();
        }
        catch (Exception e)
        {
            System.out.format("Delete exception.\nError message: %s.", e.getMessage());
        }
    }
}

File: python/PythonFileCommands/DeleteCommand.py
from Abstractions.AbstractUnaryFileCommand import *
from Validators.FileExistenceValidator import *


class DeleteCommand(AbstractUnaryFileCommand):
    @staticmethod
    def _file_existence_message():
        return "the file doesn't exist."

    @classmethod
    def usage_string(cls):
        return "delete command deletes <file1>.\n" + \
               "Use it the following way:\n" + \
               "\t'delete <file1>'\n" + \
               "\t" + cls._note_preface() + cls._argument_count_mismatch_message()

    def __init__(self, args):
        super().__init__(args)
        self._delete_validation()

    def _delete_validation(self):
        self._validation.add_validator(FileExistenceValidator(self._file, self._file_existence_message()))

    def execute(self):
        if not self._validation.validate():
            return
        try:
            self._file.unlink()
        except Exception as err:
            print("Delete exception.\nError message: {0}".format(err))



